app/lib/service/download_counts/download_counts.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:basics/basics.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:pub_semver/pub_semver.dart';
+part 'download_counts.g.dart';
+
+@JsonSerializable(includeIfNull: false)
+class CountData {
+  // We store at most two years of data, one data point per day.
+  static const countsCapacity = 731;
+  final versionCounts = <({String versionRange, List<int> counts})>[];
```

Comment:
What is `versionCounts`?

Reply:
A separate list for prereleases, but I was considering removing it. Let me do that for now.